include/deal.II/fe/fe_enriched.h

```diff
@@ -644,6 +648,231 @@ class FE_Enriched : public FiniteElement<dim, spacedim>
                           internal::FEValuesImplementation::FiniteElementRelatedData<dim,spacedim>    &output_data) const;
 };
 
+
+
+/**
+ * This namespace consists of a class needed to create a collection
+ * of finite elements (hp::FECollection) to be used with hp::DoFHandler
+ * in a domain with multiple, possibly overlapping, sub-domains with individual
+ * enrichment functions.
+ *
+ * To create hp::FECollection, graph coloring algorithm is used to assign
+ * colors to enrichment functions before creating hp::FECollection. Hence the
+ * name.
+ */
+namespace ColorEnriched
+{
+  /**
+   * An alias template for predicate function which returns a
+   * boolean for a Triangulation<dim>::cell_iterator object.
+   *
+   * This is used by helper functions and in the implementation of
+   * ColorEnriched::Helper class.
+   */
+  template <int dim>
+  using predicate_function
+    = std::function< bool
+      (const typename Triangulation<dim>::cell_iterator &) >;
+
+  /**
+   * ColorEnriched::Helper class creates a collection of finite elements
+   * (hp::FECollection) to be used with hp::DoFHandler in a domain
+   * with multiple, possibly overlapping, sub-domains with individual
+   * enrichment functions. Note that the overlapping regions may have
+   * multiple enrichment functions associated with it. This is implemented
+   * using a general constructor of FE_Enriched object which allows different
+   * enrichment functions.
+   *
+   * Construction of the class needs a base FiniteElement @p fe_base,
+   * an enriched FiniteElement @p fe_enriched, a vector of predicate
+   * functions that define subdomains as well as the corresponding enrichment
+   * functions. The class can then, by using a member
+   * function, set(hp::DofHandler &), modify active FE indices of mesh cells.
+   * The collection of finite elements to be used with hp::DoFHandler can
+   * be retrieved using a member function get_fe_collection.
+   *
+   * Consider a domain with three subdomains defined by the predicate functions.
+   * Different cells are associated with FE indices as shown in the following
+   * image.
+   * \htmlonly <style>div.image img[src="3source_fe_indices.png"]{width:25%;}</style> \endhtmlonly
+   * @image html 3source_fe_indices.png "Active FE indices"
```

Comment:
please continue paragraph by explaining connectivity graph  graph and its partitioning which ultimately leads to how we populate FECollection